adapters/handlers/rest/handlers_nodes.go

```diff
@@ -54,6 +54,32 @@ func (s *nodesHandlers) getNodesStatus(params nodes.NodesGetParams, principal *m
 	return nodes.NewNodesGetOK().WithPayload(status)
 }
 
+func (s *nodesHandlers) getNodesStatusByClass(params nodes.NodesGetClassParams, principal *models.Principal) middleware.Responder {
+	nodeStatuses, err := s.manager.GetNodeStatus(params.HTTPRequest.Context(), principal, params.ClassName)
+	if err != nil {
+		switch err.(type) {
+		case errors.Forbidden:
+			return nodes.NewNodesGetClassForbidden().
+				WithPayload(errPayloadFromSingleErr(err))
+		case enterrors.ErrUnprocessable:
+			return nodes.NewNodesGetClassUnprocessableEntity().
+				WithPayload(errPayloadFromSingleErr(err))
+		case enterrors.ErrNotFound:
+			return nodes.NewNodesGetClassNotFound().
+				WithPayload(errPayloadFromSingleErr(err))
+		default:
+			return nodes.NewNodesGetClassInternalServerError().
+				WithPayload(errPayloadFromSingleErr(err))
+		}
+	}
+
+	status := &models.NodesStatusResponse{
+		Nodes: nodeStatuses,
+	}
+
+	return nodes.NewNodesGetOK().WithPayload(status)
```

Comment:
`getNodesStatus` and `getNodesStatusByClass` share the same error handling code. could be extracted to common method

Reply:
fixed